operators/src/upgrade/controller/reconciler.rs

```diff
@@ -521,3 +522,35 @@ pub async fn start_upgrade_worker() {
         })
         .await;
 }
+
+/// Function to find whether any node drain is in progress.
+pub async fn is_draining() -> Result<bool, Error> {
+    let mut is_draining = false;
+    match UpgradeConfig::get_config()
+        .rest_client()
+        .nodes_api()
+        .get_nodes()
+        .await
```

Comment:
Can we get the node list with pagination? This could time out for very large node lists.

Reply:
Sadly we have no pagination on the nodes get...
Please add enhancement ticket :D